src/Raven.Server/Smuggler/Documents/StreamDestination.cs

```diff
@@ -683,13 +681,23 @@ public void WriteDocument(DocumentItem item, SmugglerProgressBase.CountsWithLast
                         Writer.WriteComma();
                     First = false;
 
-                    document.EnsureMetadata(_modifier);
+                    Writer.WriteStartObject();
+                    Writer.WriteDocumentPropertiesWithoutMetadata(_context, document, out BlittableJsonReaderObject metadata);
 
-                    using (var old = document.Data)
-                    using (var data = _context.ReadObject(document.Data, document.Id))
+                    if (metadata != null)
                     {
-                        _context.Write(Writer, data);
+                        Writer.WriteComma();
+                        Writer.WriteMetadata(document, metadata, s =>
+                        {
```

Comment:
I think it actually might be better to do this outside. That is, create 4 methods to do this filter.
Select between them based on the `_OperateOnTypes` in the ctor, and then pass it in.
That would save some indirection in a hot code path

Reply:
done, please check if I understood correctly.